### PR TITLE
Fixes for various JDBC issues

### PR DESCRIPTION
**Pull request by kbastani**
## Description

There are various JDBC issues that were detected through Tableau integration tests. This issue details those issues to be fixed for 0.9.0 release.

- The default maximum limit on records is added to SQL queries without a defined limit, which causes errors to be returned for some queries.
- Various timeout issues were detected in the async HTTP client requests to the Pinot broker.

These fixes address issues here:

https://github.com/apache/pinot/issues/7783

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes

Does this PR otherwise need attention when creating release notes? Things to consider:
* [ ] Yes

## Documentation

https://docs.pinot.apache.org/integrations/tableau

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
We got several magic numbers for query related timeouts. Can we make them constants so that they are easier to manage in the future?

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java [hunk 1]
```diff
@@ -209,7 +222,7 @@ public boolean isDone() {
     public ResultSetGroup get()
         throws InterruptedException, ExecutionException {
       try {
-        return get(1000L, TimeUnit.DAYS);
+        return get(200000L, TimeUnit.NANOSECONDS);
```
[the remark this answers is not in the document]
Comment (kbastani) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java:225]: I'll get the timeouts sorted out and submit the change. Thanks, @Jackie-Jiang! Good catch.

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotStatement.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
   private ResultSetGroup _resultSetGroup;
   private boolean _closed;
   private ResultSet _resultSet;
-  private int _maxRows = Integer.MAX_VALUE;
+  private int _maxRows = 1000000;
```
[the remark this answers is not in the document]
Comment (kbastani) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotStatement.java:37]: The JDBC client checks for a provided `LIMIT` statement in a SQL query and appends the `_maxRows` limit if the query did not have one. We should check to see why the Pinot broker is failing for `Integer.MAX_VALUE`.

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java [hunk 3]
```diff
@@ -167,7 +170,7 @@ public boolean isDone() {
     @Override
     public BrokerResponse get()
         throws ExecutionException {
-      return get(1000L, TimeUnit.DAYS);
+      return get(60000L * 20L, TimeUnit.NANOSECONDS);
```
[the remark this answers is not in the document]
Comment (kbastani) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java]: Right you are!

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java [hunk 4]
```diff
@@ -65,14 +65,17 @@ public JsonAsyncHttpPinotClientTransport(Map<String, String> headers, String sch
       builder.setSSLContext(sslContext);
     }
 
+    builder.setReadTimeout(60000 * 5);
```
[the remark this answers is not in the document]
Comment (kbastani) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java]: I've lowered the connect timeout to 2 seconds.

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotStatement.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotStatement.java:37]: Does the query timeout or throw exception? By default there should be no limit on this:
```
    public static final int DEFAULT_BROKER_QUERY_RESPONSE_LIMIT = Integer.MAX_VALUE;
```

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseResultSet.java [hunk 5]
```diff
@@ -229,7 +229,7 @@ public int getFetchSize()
   @Override
   public void setFetchSize(int rows)
       throws SQLException {
-    throw new SQLFeatureNotSupportedException();
+
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseResultSet.java:232]: Why do we need to remove the exception?

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseStatement.java [hunk 6]
```diff
@@ -110,7 +110,7 @@ public int getFetchSize()
   @Override
   public void setFetchSize(int rows)
       throws SQLException {
-    throw new SQLFeatureNotSupportedException();
+
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseStatement.java:113]: Why do we need to remove the exception?

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java [hunk 7]
```diff
@@ -44,7 +44,7 @@
   }
 
   PinotConnection(String controllerURL, PinotClientTransport transport, String tenant,
-      PinotControllerTransport controllerTransport) {
+                  PinotControllerTransport controllerTransport) {
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java:47]: (minor) this does not align with the `Pinot Style` format

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java [hunk 8]
```diff
@@ -117,7 +123,7 @@ public BrokerResponse executeQuery(String brokerAddress, String query)
   public BrokerResponse executeQuery(String brokerAddress, Request request)
       throws PinotClientException {
     try {
-      return executeQueryAsync(brokerAddress, request).get();
+      return executeQueryAsync(brokerAddress, request).get(600000L, TimeUnit.MILLISECONDS);
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java:126]:
```suggestion
      return executeQueryAsync(brokerAddress, request).get(BROKER_READ_TIMEOUT, TimeUnit.MILLISECONDS);
```

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java [hunk 9]
```diff
@@ -47,6 +47,9 @@
   private final Map<String, String> _headers;
   private final String _scheme;
 
+  private static final long BROKER_READ_TIMEOUT = 1200000L;
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java:50]: 20 minutes timeout seems too long. I feel 1 minute might be enough?

Comment (Jackie-Jiang) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java:50]: Also suggest appending `_MS` to be more clear (`BROKER_READ_TIMEOUT_MS` and `BROKER_CONNECT_TIMEOUT_MS`)

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java:225]: This is still not fixed

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**